chapters/analysis.typ

```diff
@@ -102,102 +197,81 @@ For other scenarios multiple refactoring operations already exist (e.g. switchin
+The LLVM project strictly adheres to a well-defined architecture for testing. 
+To align with project guidelines @clangd_testing, automated unit tests must be authored prior to the acceptance of any code contributions. 
 The name of these files is usually the name of the class itself and use the googletest framework @googletest_framework.
 
-Unit tests are added to `llvm-project/clang-tools-extra/clangd/unittests`
+Unit tests are added to `llvm-project/clang-tools-extra/clangd/unittests`.
 
 // TODO: explain test concept of clangd (for tweaks)
 
-#pagebreak()
+For testing tweaks four functions are typically used `EXPECT_THAT`, `EXPECT_AVAILABLE`, `EXPECT_UNAVAILABLE`.
 
-=== Code Actions
-// Kommentar Jeremy: Was isch "LSP system"?
-// Kommentar Jeremy: C++ het kei interfaces
-Tweaks supply the majority of code actions in the LSP system. These compact plugins implement the Tweak interface and reside in the `refactor/tweaks` directory, where they are registered through the linker. 
-When presented with a selection, they can swiftly assess if they are applicable and, if necessary, create the edits, potentially at a slower pace.
-These fundamental components constitute the foundation of the LSP code-actions flow.
+/ `EXPECT_THAT` : #[
+Executes the `apply` function and compares the result with the expected code.
+]
 
-Each tweak has it's own class and is structured as follows:
-
-// Kommentar Jeremy: Code size vellecht chli chlinner mache
-// Kommentar Jeremy: Listing
-
-#[
-  #set text(size: 0.8em)
-  ```cpp
-  namespace clang {
-  namespace clangd {
-  namespace {
-  /// Feature description
-  class RefactorExample : public Tweak {
-  public:
-    const char *id() const final;
-
-    bool prepare(const Selection &Inputs) override;
-    Expected<Effect> apply(const Selection &Inputs) override;
-    std::string title() const override { return "A Refactoring Example"; }
-    llvm::StringLiteral kind() const override {
-      return CodeAction::REFACTOR_KIND;
-    }
-  };
+/ `EXPECT_AVAILABLE` : #[
+Check if the refactoring feature is available on certain cursor position within the code. 
+The `prepare` function is executed.
+]
 
-  REGISTER_TWEAK(RefactorExample)
+/ `EXPECT_UNAVAILABLE` : #[
+Check if the refactoring feature is unavailable on certain cursor position within the code. 
+The `prepare` function is executed.
+]
 
-  bool RefactorExample::prepare(const Selection &Inputs) {
-    // Check if refactoring is possible
-  }
+=== Code Actions
 
-  Expected<Tweak::Effect> RefactorExample::apply(const Selection &Inputs) {
-    // Refactoring code
-  }
+All refactoring features so called "tweaks" reside in the `refactor/tweaks` directory, where they are registered through the linker. 
+The return type of a tweak is always a Code Action, which represents a change that can be performed in code. @code_action
+These compact plugins inherit the Tweak class which acts as a interface base.
```

Comment:
```suggestion
These compact plugins inherit the Tweak class which acts as a base interface.
```

Reply:
ide klass staht " An interface base for small context-sensitive refactoring actions." als kommentar also würis so lah